app/controllers/course/forum/posts_controller.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Forum::PostsController < Course::Forum::Controller
+  include Course::Discussion::PostsConcern
+
+  protected
+
+  def default_return_path
+    course_forum_topic_path(current_course, @forum, @topic)
+  end
+
+  def auto_subscribe_topic
+    true
+  end
+
+  def discussion_topic
```

Comment:
I think I only need to load and authorize the concrete topic in this method, because this method will be called before any action including those are in the PostsConcern.